### PR TITLE
Add nullability and generics support

### PR DESCRIPTION
**Pull request by Grubas7**

### CODE REVIEW

**Comment by Grubas7** [comment]
@jhersh ready for review 🙂

File: SSDynamicText/NSAttributedString+SSTextSize.h [hunk 1]
```diff
@@ -3,10 +3,10 @@
 //  SSDynamicText
 //
 //  Created by Remigiusz Herba on 28/08/15.
-//
+//  Copyright (c) 2015 Splinesoft. All rights reserved. 
 //
 
-#import <Foundation/Foundation.h>
+@import Foundation;
```
Comment (jhersh) [inline SSDynamicText/NSAttributedString+SSTextSize.h]: Modules are :+1: but the Podfile might need an update. See [this example](https://github.com/splinesoft/SSDataSources/blob/master/SSDataSources.podspec#L14).

File: SSDynamicText/SSDynamicLabel.m [hunk 2]
```diff
@@ -67,14 +49,33 @@ - (void)dealloc {
...
+    return _textSizeChanger;
+}
+
 - (SSDynamicTextSizeChanger *)createTextChanger {
     SSDynamicTextSizeChanger *changer = [[SSDynamicTextSizeChanger alloc] init];
     __weak typeof(self) weakSelf = self;
 
     changer.fontChangeBlock = ^(UIFont *font) {
-        [super setFont:font];
+        super.font = font;
```
Comment (jhersh) [inline SSDynamicText/SSDynamicLabel.m]: A minor stylistic thing, but this is one area where I don't tend to use dot syntax. `[super setFont:font]` feels a bit more explicit about which selector we're invoking.

File: SSDynamicText/NSAttributedString+SSTextSize.m [hunk 3]
```diff
@@ -3,7 +3,7 @@
 //  SSDynamicText
 //
 //  Created by Remigiusz Herba on 28/08/15.
-//
+//  Copyright (c) 2015 Splinesoft. All rights reserved.
```
Comment (jhersh) [inline SSDynamicText/NSAttributedString+SSTextSize.m]: 2016? :)

Comment (Grubas7) [inline SSDynamicText/NSAttributedString+SSTextSize.m]: I copied date from lines above.
In this files date is `28/08/15` ;)

File: SSDynamicText/SSDynamicLabel.m (hunk 2 above)
Comment (Grubas7) [inline SSDynamicText/SSDynamicLabel.m]: Reverted.

File: SSDynamicText/NSAttributedString+SSTextSize.h (hunk 1 above)
Comment (Grubas7) [inline SSDynamicText/NSAttributedString+SSTextSize.h]: Right now `SSDynamicLabel`, `SSDynamicTextView` and `SSDynamicTextField` have `@import` and everything works.
Could you write example when you had a problem with pod without `-fmodules` compiler flag?

Thanks in advance!

Comment (jhersh) [inline SSDynamicText/NSAttributedString+SSTextSize.h]: CocoaPods' generated library doesn't have the module flag enabled, so adding this flag in the podspec ensures that modules are enabled for the library. I haven't checked in a while; perhaps CocoaPods has changed or improved here?

Comment (Grubas7) [inline SSDynamicText/NSAttributedString+SSTextSize.h]: Look like cocoapods changed this flag to enabled by default. But I added to be proof for future changes.

**Pull request merged**